code/game/machinery/doors/door.dm

```diff
@@ -239,6 +243,30 @@
 	if(HAS_TRAIT(src, TRAIT_CMAGGED) && I.can_clean()) //If the cmagged door is being hit with cleaning supplies, don't open it, it's being cleaned!
 		return
 
+	if(istype(I, /obj/item/stack/sheet/wood) && user.a_intent == INTENT_HELP)
+		var/obj/item/stack/sheet/wood/S = I
+		if(!density)
+			to_chat(user, "<span class='warning'>\The [src] must be closed!</span>")
+			return
+		if(S.get_amount() < 2)
+			to_chat(user, "<span class='warning'>You need at least 2 planks of wood to barricade this!</span>")
+			return
+		if(door_barricaded)
+			to_chat(user, "<span class='warning'>There's already a barricade here!</span>")
+			return
+		if((/mob) in get_turf(src))
```

Comment:
this will catch ghosts and other invisible creatures. Making this `mob/living` and check density of the mob or something so this doesn't catch hidden antags or something